asv_bench/benchmarks/utils/common.py

```diff
@@ -171,38 +159,26 @@ def gen_str_int_data(nrows: int, ncols: int, rand_low: int, rand_high: int) -> d
         Low bound for random generator.
     rand_high : int
         High bound for random generator.
+    cache_key : tuple
+        Key to store generated data in cache.
 
     Returns
     -------
     dict
         Number of keys - `ncols`, each of them store np.ndarray of `nrows` length.
         One of the columns with string values.
     """
-    cache_key = ("str_int", nrows, ncols, rand_low, rand_high)
-    if cache_key in data_cache:
-        return data_cache[cache_key]
-
-    logging.info(
-        "Generating str_int data {} rows and {} columns [{}-{}]".format(
-            nrows, ncols, rand_low, rand_high
-        )
-    )
-    data = gen_int_data(nrows, ncols, rand_low, rand_high).copy()
+    data = gen_data("int", nrows, ncols, rand_low, rand_high).copy()
```

Comment:
Why call a higher level function from a lower level function? `gen_int_data` will be called anyway, right?

Reply:
If we call gen_int_data the data wouldnt be cached as the caching logic has been moved to get_data, Thus I figured calling gen_data higher level function would be better to cache data for later.

Reply:
> If we call gen_int_data the data wouldnt be cached as the caching logic has been moved to get_data, 

That's ok, we shouldn't use a low level function if there is a way to use it through a high level function.

Reply:
In fact, we should leave this line as it was before, imagine the execution flow, we design it so that it starts with function `gen_data`, since it contains additional caching functionality. If the function of the lower part of the flow (for example, `gen_str_int_data`) calls the upper one, then the flow is not unidirectional (this is usually more difficult to work with and understand).

Reply:
Got it, Done

Reply:
Thanks!